src/openms/source/QC/DBSuitability.cpp

```diff
@@ -597,6 +616,78 @@ namespace OpenMS
     return -(db_slope) / (deNovo_slope);
   }
 
+  UInt DBSuitability::numberOfUniqueProteins_(const std::vector<PeptideIdentification>& peps, UInt number_of_hits) const
+  {
+    set<String> proteins;
+
+    for (const auto& pep : peps)
+    {
+      const vector<PeptideHit>& hits = pep.getHits();
+      for (UInt i = 0; i < number_of_hits; ++i)
+      {
+        const PeptideHit& hit = hits[i];
+
+        // skip if the hit is a decoy hit
+        if (!hit.metaValueExists("target_decoy"))
+        {
+          throw(Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No target/decoy information found! Make sure 'PeptideIndexer' is run beforehand."));
+        }
+        if (hit.getMetaValue("target_decoy") == "decoy") continue;
+
+        // insert protein accessions
+        const set<String> accessions = hit.extractProteinAccessionsSet();
+        for (const String& acc : accessions)
+        {
+          String acc_copy(acc);
+          if (acc.find(Constants::UserParam::CONCAT_PEPTIDE) != String::npos) // skip novo accessions
+          {
+            continue;
+          }
+          if (boost::regex_search(acc_copy.toLower(), decoy_pattern_)) // skip decoy accessions
```

Comment:
ok, that;s why you copy...
but then maybe do `if (boost::regex_search(String(acc).toLower(), decoy_pattern_))`, to make the copy later